### PR TITLE
Move `RelatedManager` to `django.db.models.fields.related_descriptors`

### LINKED ISSUE #1454: Add ManyRelatedManager and distinguish from RelatedManager
Methods of `RelatedManager` and `ManyRelatedManager` have different signatures.

I noticed while reviewing #1443.

### PR DESCRIPTION
**Pull request by flaeppe**
Aligns with the module `RelatedManager` is locally declared

## Related issues

Closes #1454

### CODE REVIEW

File: django-stubs/db/models/fields/related_descriptors.pyi [hunk 1]
```diff
@@ -86,6 +86,18 @@ class ReverseManyToOneDescriptor:
     def __get__(self, instance: Model, cls: Any = ...) -> type[RelatedManager[Any]]: ...
     def __set__(self, instance: Any, value: Any) -> NoReturn: ...
 
+class RelatedManager(BaseManager[_M], Generic[_M]):
```
Comment (intgr) [inline django-stubs/db/models/fields/related_descriptors.pyi:90]: I think a comment to the fact that this doesn't directly correspond to a runtime class is useful. Feel free to tweak.

```suggestion
# Fake class, Django defines RelatedManager inside a function body
class RelatedManager(BaseManager[_M], Generic[_M]):
```

Would make sense to add the same for `ManyRelatedManager` as well

**Review by intgr — APPROVED** [review]
:+1: Makes sense to move it to `related_descriptors` where Django itself defines it.

File: django-stubs/db/models/fields/related_descriptors.pyi (hunk 1 above)
Comment (flaeppe) [reply django-stubs/db/models/fields/related_descriptors.pyi:90]: Am thinking to copy the comment from the `allowlist.txt` where these classes are also present

> 'RelatedManager' does exist and is declared locally, inside a function body

As well as

> 'ManyRelatedManager' does exist and is declared locally, inside a function body

Comment (flaeppe) [reply django-stubs/db/models/fields/related_descriptors.pyi:90]: Nvm, that didn't sound so good. Your suggestion reads better

File: django-stubs/db/models/fields/related_descriptors.pyi [hunk 2]
```diff
@@ -86,6 +86,19 @@ class ReverseManyToOneDescriptor:
     def __get__(self, instance: Model, cls: Any = ...) -> type[RelatedManager[Any]]: ...
     def __set__(self, instance: Any, value: Any) -> NoReturn: ...
 
+# Fake class, Django defines 'RelatedManager' inside a function body
+class RelatedManager(BaseManager[_M], Generic[_M]):
```
Comment (intgr) [inline django-stubs/db/models/fields/related_descriptors.pyi:90]: You also changed the base class from `Manager` to `BaseManager`.

I suppose it's more correct (managers *usually* derive from `Manager`, but not always)? But this causes several new inspections in my projects. Seems like something that should be mentioned in release notes.

Comment (intgr) [reply django-stubs/db/models/fields/related_descriptors.pyi:90]: Hmm, I think we should export `RelatedManager` and `ManyRelatedManager` classes from `django-stubs-ext` to make their usage more convenient.

Right know if anyone wants to use them in type hints, they have to hide the import behind an `if TYPE_CHECKING:` condition and use quoting for the type, e.g. `blah_set: "RelatedManager[Blah]"`.

Comment (flaeppe) [reply django-stubs/db/models/fields/related_descriptors.pyi:90]: I changed base to align the 2 classes. But any of them is probably fine here really.

And yes, I agree, we should export them.

**Pull request merged**